models/tenant.py

```diff
@@ -36,7 +40,9 @@ def json(self):
             'phone': self.phone,
             'propertyID': self.propertyID,
             'propertyName': self.property.name if self.property else None,
-            'staff': [user.json() for user in self.staff] if self.staff else None
+            'staff': [user.json() for user in self.staff] if self.staff else None,
+            'unit': self.unit,
+            'addedOn': self.addedOn
```

Comment:
This is causing the majority of the failures. A date object is not serializable. It needs to be converted to a string.
`self.addedOn.strftime("%m/%d/%Y, %H:%M:%S")`

Reply:
This is resolved

Reply:
The strftime method was placed in the initializer. It should be here in the json method. We want to store the date in the db as a date object, and when we send to front end it should be the string representation.

After making these changes and running tests locally you should then see 2 failing tests. Those also will need to be resolved. We can pair on it if you like.